modules/community-algorithms/pages/k-means-clustering.adoc

```diff
@@ -0,0 +1,51 @@
+= k-Means Clustering
+
+In the k-Means Clustering algorithm, vertices are gradually sorted into an arbitrary number _k_ of clusters based on averaging their properties.
+
+First, _k_ vertices are selected at random to act as centroids, or the center of their clusters. The first clusters are defined as consisting of the vertices closest to those centroids. These clusters are refined by calculating the mean position of their points, then redrawing the boundaries of the clusters by finding the closest vertices to those mean position points.
+
+The last two steps are repeated many more times until each iteration no longer changes the location of the centroids or the vertices belonging to the clusters.
+
+Vertices in the graph need to have embeddings for this algorithm to work, because the distances are calculated based on the embeddings rather than any other factor.
```

Comment:
Should specifically call out the fact that they need to modify certain lines in the algorithm. If I was a new user coming onto this page, I probably would not understand what it means to "have embeddings".